server/src/main/java/org/apache/druid/client/CachingClusteredClient.java

```diff
@@ -385,6 +385,7 @@ private String computeCurrentEtag(final Set<ServerToSegment> segments, @Nullable
           break;
         }
         hasher.putString(p.getServer().getSegment().getId().toString(), StandardCharsets.UTF_8);
+        hasher.putString(p.rhs.getInterval().toString(), StandardCharsets.UTF_8);
```

Comment:
Comment here would be nice, explaining that this interval is the query interval, not the segment interval, and it's important for it to be part of the cache key.